os/fs/driver/mtd/smart.c

```diff
@@ -1875,36 +1797,16 @@ static int smart_scan(FAR struct smart_struct_s *dev)
 		}
 
 		/* Test if this sector has been committed. */
-
-		if (!status_committed) {
-			/* Confirm that this sector is truly blank.*/
-			if (logicalsector < dev->totalsectors) {
-				/* This sector if not erased and not committed, might have been corrupted. Commit and release this. */
-#if CONFIG_SMARTFS_ERASEDSTATE == 0xFF
-				header.status = header.status & ~(SMART_STATUS_COMMITTED | SMART_STATUS_RELEASED);
-#else
-				header.status = header.status | SMART_STATUS_COMMITTED | SMART_STATUS_RELEASED;
-#endif
-				status_committed = true;
-				status_released = true;
-				ret = smart_bytewrite(dev, readaddress + offsetof(struct smart_sect_header_s, status), 1, &header.status);
-				if (ret < 0) {
-					fdbg("checkpoint line %d\n", __LINE__);
-					goto err_out;
-				}
-			} else {
-				continue;
-			}
-		}
-
-		/* This block is now commited, therefore not free. Update the erase block's freecount.*/
+		if (status_committed) {
+			/* This block is now commited, therefore not free. Update the erase block's freecount.*/
 
 #ifdef CONFIG_MTD_SMART_PACK_COUNTS
-		smart_add_count(dev, dev->freecount, sector / dev->sectorsPerBlk, -1);
+			smart_add_count(dev, dev->freecount, sector / dev->sectorsPerBlk, -1);
```

Comment:
- Creating the sector mappings and calculating free/released sectors can stay with smart_scan for general cases.